### ISSUE TITLE
Total payments problem

### ISSUE THREAD

**Issue by strengzs** [ChristopherKlay/StadiaEnhanced#110]
Hi, 
Something is wrong with the display of total payments specially at Hungarian localization settings. 
English is better, but not perfect.

To get the situation more complex 
- Hungarian decimal signal is: "," - comma
- English decimal signal is: "." - dot

However we can leave with "." dot as a decimal signal. No problem, but in this case do not use "," for segmentation in the number, because that will confusing. (space is not mandatory for segmentation, but looks better)

Hungarian
![image](https://user-images.githubusercontent.com/77846114/112139045-91c60980-8bd2-11eb-85bd-6fd780ca208b.png)

English
![image](https://user-images.githubusercontent.com/77846114/112138798-3bf16180-8bd2-11eb-8cba-89019ce75b2d.png)

**Comment by ChristopherKlay** [comment]
Yea, the feature has some issues due to having to support a ton of different currencies and no way to check which one is used automatically (i.e. via a language code like 'EUR').

In this case, would `Total 10751.00 637.00 Ft` be better?

**Comment by strengzs** [comment]
I understand, sorry for my language and currency complications. ;)
`Total 10751.00 Ft` is good. `Total 10 751.00 Ft` is the perfect.
I do not get where 637.00 comes from anyway...

English is wrong as well with the 2 "." dots. - Problem comes from decimal signal interpretation.
Thank you!

**Comment by ChristopherKlay** [comment]
No need to be sorry - definitely working on fixing these!

Could you open up the payments page and copy this into the console (F12) to see if the new result works fine?

```
var enhanced_purchases = document.querySelectorAll('.o2NwGe')
if (enhanced_purchases.length != 0) {
	var enhanced_currency = document.querySelector('.o2NwGe').textContent.replace(/[0-9]+[,. ]+[0-9]*[,. ]*[0-9]+/, 'CUR')
	var enhanced_payments = [];

	for (var i = 0; i < enhanced_purchases.length; i++) {
		enhanced_payments.push(parseFloat(enhanced_purchases[i].textContent.replace(/[^0-9,.]/g, '').replace(",", ".")))
	}

	var enhanced_paytotal = enhanced_payments.reduce((a, b) => a + b, 0).toFixed(2)
	enhanced_paytotal = "Total " + enhanced_currency.replace('CUR', enhanced_paytotal)

	alert(enhanced_paytotal)
}
```
The script will calculate it using a different pattern (that includes spaces, which created a issue) and display the total value via a prompt.

**Comment by strengzs** [comment]
HUN:
![image](https://user-images.githubusercontent.com/77846114/112154739-da86be00-8be4-11eb-86c0-759e0749f204.png)
ENG:
![image](https://user-images.githubusercontent.com/77846114/112154874-0013c780-8be5-11eb-9413-ae0d75b2f504.png)

**Comment by ChristopherKlay** [comment]
Alright that.. definitely went in the right direction when it comes to the space issue at the very least.

Could you copy me the output of this (rightclick on the array in the console, copy object);
```
var enhanced_purchases = document.querySelectorAll('.o2NwGe')

if (enhanced_purchases.length != 0) {
	var enhanced_payments = [];

	for (var i = 0; i < enhanced_purchases.length; i++) {
		enhanced_payments.push(enhanced_purchases[i].textContent)
	}
    
    console.dir(enhanced_payments)
}
```
It effectively contains the full text content that's supposed to be filtered afterwards and would allow me to check/fix the regex filter more quickly (:

**Comment by strengzs** [comment]
I was not able to copy. It says undefined, so I do not know where to rightclick and copy, but I made `alert(enhanced_payments)`, so I hope you need this:
HUN:  `4 637,00 Ft,4 497,00 Ft,1 617,00 Ft,0,00 Ft`
![image](https://user-images.githubusercontent.com/77846114/112163746-9946dc00-8bed-11eb-86ee-15d6c4fc566c.png)

ENG:   `Ft4,637.00,Ft4,497.00,Ft1,617.00,Ft0.00`
![image](https://user-images.githubusercontent.com/77846114/112163916-c2676c80-8bed-11eb-86ae-bdc12bbaf2c0.png)

**Comment by ChristopherKlay** [comment]
That was perfectly fine!

There was another issue, that's why i tested via the (heavily overcomplicated) regex above. Granted this would be easier if i could test on the actual page - but this *should* work now;

```
var enhanced_purchases = document.querySelectorAll('.o2NwGe')
if (enhanced_purchases.length != 0) {
	var enhanced_currency = document.querySelector('.o2NwGe').textContent.replace(/[0-9,.]+/, 'CUR')
	var enhanced_payments = [];

	for (var i = 0; i < enhanced_purchases.length; i++) {
		if (enhanced_purchases[i].textContent.match(/[,].*[.]/)) {
			enhanced_payments.push(parseFloat(enhanced_purchases[i].textContent.replace(/[^0-9,. ]+/, '').replace(",", "")))
		} else {
			enhanced_payments.push(parseFloat(enhanced_purchases[i].textContent.replace(/[^0-9,. ]+/, '').replace(",", ".")))
		}
	}

	var enhanced_paytotal = enhanced_payments.reduce((a, b) => a + b, 0).toFixed(2)
	enhanced_paytotal = "Total " + enhanced_currency.replace('CUR', enhanced_paytotal)

	console.log(enhanced_paytotal)
}
```
It's effectively bypassing the whole "," and "." issue by only dropping the "," if both exist.

**Comment by strengzs** [comment]
HUN is still broken as at the beginning. 637 is displayed as well, but I have no idea why.
![image](https://user-images.githubusercontent.com/77846114/112200314-67df0800-8c0f-11eb-9435-ceab30f4c99f.png)

**ENG is good now:**
![image](https://user-images.githubusercontent.com/77846114/112199680-b0e28c80-8c0e-11eb-8db3-1a57485a249c.png)

**Comment by ChristopherKlay** [comment]
Found the (hopefully) last issue; The currency calculation was checking for spaces, but the currency string itself wasn't checked for them, which resulted in "673,00 Ft" being saved as the currency, not just " Ft".

```
var enhanced_purchases = document.querySelectorAll('.o2NwGe')
if (enhanced_purchases.length != 0) {
	var enhanced_currency = document.querySelector('.o2NwGe').textContent.replace(/[0-9].*[0-9]/, 'CUR')
	var enhanced_payments = [];

	for (var i = 0; i < enhanced_purchases.length; i++) {
		if (enhanced_purchases[i].textContent.match(/[,].*[.]/)) {
			enhanced_payments.push(parseFloat(enhanced_purchases[i].textContent.replace(/[^0-9,. ]+/, '').replace(",", "")))
		} else {
			enhanced_payments.push(parseFloat(enhanced_purchases[i].textContent.replace(/[^0-9,. ]+/, '').replace(",", ".")))
		}
	}

	var enhanced_paytotal = enhanced_payments.reduce((a, b) => a + b, 0).toFixed(2)
	enhanced_paytotal = "Total " + enhanced_currency.replace('CUR', enhanced_paytotal)

	console.log(enhanced_paytotal)
}
```

**Comment by strengzs** [comment]
Finally you did it. Works fine, thank you!
HUN:
![image](https://user-images.githubusercontent.com/77846114/112205966-ad9ecf00-8c15-11eb-80f3-2dd167a21997.png)

ENG:
![image](https://user-images.githubusercontent.com/77846114/112206428-3453ac00-8c16-11eb-8a80-8fe679a4aafa.png)

(It will be fine for now, thank you again. To implement the proper decimal symbol, digit grouping, digit grouping symbol could be very complicated, because all of them are different for Hungarian and English, and could vary for other languages as well, so leave it as it is now.)

**Comment by ChristopherKlay** [comment]
This should be fixed in `1.10.1` (:

**Comment by strengzs** [comment]
Thank you. Its is OK. Confirmed. (If you have a lot of time you can add digit grouping with spaces  like. 12 345.00 EUR)
